secure/storage/src/tests/suite.rs

```diff
@@ -1,110 +1,180 @@
 // Copyright (c) The Libra Core Contributors
 // SPDX-License-Identifier: Apache-2.0
 
-use crate::{Error, Policy, Storage, Value};
+use crate::{tests::vault, Error, InMemoryStorage, OnDiskStorage, Policy, Storage, Value};
 use libra_crypto::{ed25519::Ed25519PrivateKey, Uniform};
+use libra_temppath::TempPath;
 use rand::{rngs::StdRng, SeedableRng};
 
-const KEY_KEY: &str = "key";
-const U64_KEY: &str = "u64";
-
-/// This helper function checks various features of the secure storage behaviour relating to K/V
-/// support, e.g., ensuring errors are returned when attempting to retrieve missing keys, and
-/// verifying sets and gets are implemented correctly.
-pub fn run_test_suite(storage: &mut dyn Storage, name: &str) {
-    assert!(
-        storage.available(),
-        eprintln!("Backend storage, {}, is not available", name)
-    );
+/// This suite contains tests for secure storage backends. We test the correct functionality
+/// of both key/value and cryptographic operations for storage implementations. All storage backend
+/// implementations should be tested using the tests in this suite.
+
+/// This holds the canonical list of secure storage tests. It allows different callers
+/// in the test suite to ensure they're executing all tests.
+/// Note: this is required because vault tests cannot be run in the usual fashion (i.e., vault
+/// tests rely on first running the vault docker script in `docker/vault/run.sh`). In addition,
+/// vault tests cannot currently be run in parallel, as each test uses the same vault instance.
+/// TODO(joshlind): see if we can overcome the above limitations to remove this list.
+const STORAGE_TESTS: &'static [fn(&mut dyn Storage)] = &[
+    test_ensure_storage_is_available,
+    test_create_get_set_unwrap,
+    test_create_key_value_twice,
+    test_get_set_non_existent,
+    test_verify_incorrect_value_types_unwrap,
+    test_create_get_and_test_key_pair,
+    test_create_key_pair_twice,
+    test_get_uncreated_key_pair,
+];
+
+#[test]
+fn execute_storage_tests_all_backends() {
+    for test in STORAGE_TESTS.iter() {
+        // Test the in_memory implementation
+        let mut storage = Box::new(InMemoryStorage::new());
+        test(storage.as_mut());
+
+        // Test the on_disk implementation
+        let temp_path = TempPath::new();
+        let mut storage = Box::new(OnDiskStorage::new(temp_path.path().to_path_buf()));
+        test(storage.as_mut());
+    }
+}
```

Comment:
kinda prefer these to be in their own little boring files since it is easy to see.

Reply:
Sure, I don't feel that strongly about this, so I've called the test suite in each individual file.